src/test/gtest_main.cpp

```diff
@@ -1,19 +1,55 @@
-#include <sys/stat.h>
-#include <sys/types.h>
+#if __has_include(<filesystem>)
+#include <filesystem>
+namespace filesystem = std::filesystem;
+#else
+#include <experimental/filesystem>
+namespace filesystem = std::experimental::filesystem;
+#endif
 
 #include <gtest/gtest.h>
+#include "base_test.hpp"
 
 #include "utils/assert.hpp"
 #include "utils/performance_warning.hpp"
 
+void create_test_data_directory(std::optional<std::string>& prefix) {
+  opossum::Assert(!filesystem::exists(opossum::test_data_path),
+    "Cannot create directory for test data: " + opossum::test_data_path +" - it already exists.");
```

Comment:
If I run the test and ctrl-c it / kill it / it segfaults, I cannot run it again:

```
[:~/Projekte/Opossum/Git] fix-filename-tests(+5/-6)+* ± ./clang-release/hyriseTest
libc++abi.dylib: terminating with uncaught exception of type std::logic_error: Cannot create directory for test data: ./.hyrise_test_data/ - it already exists.
```

Shouldn't the folder be cleared when we start the test suite?

Reply:
I believe not. In that case, our test program would clear any given folder. I think the cleanest solution does not clear it, but notifies the user.